libs/template/materialize.go

```diff
@@ -60,35 +84,76 @@ func Materialize(ctx context.Context, configFilePath string, templateFS fs.FS, o
 
 	// Prompt user for any missing config values. Assign default values if
 	// terminal is not TTY
-	err = config.promptOrAssignDefaultValues(r)
+	err = t.config.promptOrAssignDefaultValues(t.renderer)
 	if err != nil {
 		return err
 	}
-	err = config.validate()
+	return t.config.validate()
+}
+
+func (t *Template) printSuccessMessage(ctx context.Context) error {
+	success := t.config.schema.SuccessMessage
+	if success == "" {
+		cmdio.LogString(ctx, "✨ Successfully initialized template")
+		return nil
+	}
+
+	success, err := t.renderer.executeTemplate(success)
 	if err != nil {
 		return err
 	}
+	cmdio.LogString(ctx, success)
+	return nil
+}
+
+func (t *Template) logTelemetry(ctx context.Context) error {
+	// Only log telemetry input for Databricks owned templates. This is to prevent
+	// accidentally collecting PII from custom user templates.
+	templateEnumArgs := map[string]string{}
+	if t.IsDatabricksOwned {
+		templateEnumArgs = t.config.enumValues()
+	} else {
+		t.Name = "custom"
+	}
 
-	// Walk and render the template, since input configuration is complete
-	err = r.walk()
+	event := telemetry.DatabricksCliLog{
+		BundleInitEvent: &events.BundleInitEvent{
+			Uuid:             bundleUuid,
+			TemplateName:     t.Name,
+			TemplateEnumArgs: templateEnumArgs,
+		},
+	}
+
+	return telemetry.Log(ctx, telemetry.FrontendLogEntry{
+		DatabricksCliLog: event,
+	})
+}
+
+// This function materializes the input templates as a project, using user defined
+// configurations.
+func (t *Template) Materialize(ctx context.Context) error {
+	err := t.resolveTemplateInput(ctx)
 	if err != nil {
 		return err
 	}
 
-	err = r.persistToDisk(ctx, outputFiler)
+	// Walk the template file tree and compute in-memory representations of the
+	// output files.
+	err = t.renderer.walk()
 	if err != nil {
 		return err
 	}
 
-	success := config.schema.SuccessMessage
-	if success == "" {
-		cmdio.LogString(ctx, "✨ Successfully initialized template")
-	} else {
-		success, err = r.executeTemplate(success)
-		if err != nil {
-			return err
-		}
-		cmdio.LogString(ctx, success)
+	// Flush the output files to disk.
+	err = t.renderer.persistToDisk(ctx, t.OutputFiler)
+	if err != nil {
+		return err
 	}
-	return nil
+
+	err = t.printSuccessMessage(ctx)
+	if err != nil {
+		return err
+	}
+
+	return t.logTelemetry(ctx)
```

Comment:
It would be nice to log the status of the bundle init (success, error)

Reply:
I'll do that in a followup PR.